lms/djangoapps/program_enrollments/api/api.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+ProgramEnrollment internal api
+"""
+from __future__ import absolute_import, unicode_literals
+
+from datetime import datetime, timedelta
+from pytz import UTC
+
+from django.urls import reverse
+
+from six import iteritems
+
+from bulk_email.api import is_bulk_email_feature_enabled, is_user_opted_out_for_course
+from edx_when.api import get_dates_for_course
+from xmodule.modulestore.django import modulestore
+from lms.djangoapps.program_enrollments.api.v1.constants import (
+    CourseEnrollmentResponseStatuses,
+    CourseRunProgressStatuses,
+    MAX_ENROLLMENT_RECORDS,
+    ProgramEnrollmentResponseStatuses,
+)
+
+
+def get_due_dates(request, course_key, user):
+    """
+    Get due date information for a user for blocks in a course.
+
+    Arguments:
+        request: the request object
+        course_key (CourseKey): the CourseKey for the course
+        user: the user object for which we want due date information
+
+    Returns:
+        due_dates (list): a list of dictionaries containing due date information
+            keys:
+                name: the display name of the block
+                url: the deep link to the block
+                date: the due date for the block
+    """
+    dates = get_dates_for_course(
+        course_key,
+        user,
+    )
+
+    store = modulestore()
+
+    due_dates = []
+    for (block_key, date_type), date in iteritems(dates):
+        if date_type == 'due':
+            block = store.get_item(block_key)
+
+            # get url to the block in the course
+            block_url = reverse('jump_to', args=[course_key, block_key])
+            block_url = request.build_absolute_uri(block_url)
+
+            due_dates.append({
+                'name': block.display_name,
+                'url': block_url,
+                'date': date,
+            })
+    return due_dates
+
+
+def get_course_run_url(request, course_id):
+    """
+    Get the URL to a course run.
+
+    Arguments:
+        request: the request object
+        course_id (string): the course id of the course
+
+    Returns:
+        (string): the URL to the course run associated with course_id
+    """
+    course_run_url = reverse('openedx.course_experience.course_home', args=[course_id])
+    return request.build_absolute_uri(course_run_url)
+
+
+def get_emails_enabled(user, course_id):
```

Comment:
Maybe something like `is_bulk_email_enabled_for_user_and_course` instead.

Reply:
anything in program_enrollments is stuff i've moved to an api.py file so it's "appropriate" for us to use in another app. I didn't write this code, just reusing it. I'd like to avoid renaming it if possible